devices/emulator/impl.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2019-2020 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#include "impl.h"
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+#define MAXLINE 1024
+#define IMSI_LEN 15
+#define CBC_IV_SIZE 16
+
+static int emulator_init(void);
+static int emulator_get_key(uint8_t *);
+static int emulator_get_device_id(char *);
+static void emulator_release(void);
+static inline void register_emulator(void);
+static inline void unregister_emulator(void);
+static void write_log(char *path, char *msg, size_t msg_len);
+
+/* Setting data to produce predictable results for emulator */
+// init vector for aes
+const static uint8_t iv_global[16] = {164, 3, 98, 193, 52, 162, 107, 252, 184, 42, 74, 225, 157, 26, 88, 72};
+// device id
+const static char *device_id = "470010171566423";
+// private key
+const static uint8_t private_key[32] = {82,  142, 184, 64,  74, 105, 126, 65,  154, 116, 14,  193, 208, 41,  8,  115,
+                                        158, 252, 228, 160, 79, 5,   167, 185, 13,  159, 135, 113, 49,  209, 58, 68};
+
+extern struct uart_operations emulator_uart;
+
+static struct device_operations emulator_ops = {.init = &emulator_init,
```

Comment:
Avoiding using the form `&emulator_init`. Use `.init = emulator_init` instead. Be aware of consistent indention.

Reply:
Ready to resolve?

Reply:
Yes, I will do this tomorrow. Code review can also start.